+nosnoc/+discrete_time_problem/Stewart.m

```diff
@@ -870,21 +871,31 @@ function create_solver(obj, solver_options, plugin)
             end
 
             if ~exist('plugin')
-                plugin = 'scholtes_ineq';
+                plugin = 'reg_homotopy';
             end
 
-            obj.finalize_assignments();
+            if ~exist('regenerate')
+                regenerate = false;
+            end
+
+            if regenerate || isempty(obj.solver)
+                obj.finalize_assignments();
 
-            % Sort by indices to recover almost block-band structure.
-            obj.w.sort_by_index();
-            obj.g.sort_by_index();
+                % Sort by indices to recover almost block-band structure.
+                obj.w.sort_by_index();
+                obj.g.sort_by_index();
 
-            solver_options.assume_lower_bounds = true;
+                solver_options.assume_lower_bounds = true;
```

Comment:
if this is set here, then setting it outside in solver_options has no effect at all?

Reply:
Yes, as it should. For `nosnoc` specific problems this should _always_ be true otherwise the numerics in the relaxed NLP become nasty due to duplicate constraints.